packages/eventual-send/src/index.d.ts

```diff
@@ -257,11 +257,11 @@ interface EProxy {
    * E.when(x, res, rej) is equivalent to
    * HandledPromise.resolve(x).then(res, rej)
    */
-  readonly when: <T, U>(
+  readonly when: <T>(
```

Comment:
I think by defaulting `U` to `T` (for when neither reactions are used) might solve the problem

```suggestion
  readonly when: <T, U = T>(
```